### PR TITLE
Bump @metamask/assets-controller to v20.0.0

### PR DESCRIPTION
**Pull request by adonesky1**
Bumps `@metamask/assets-controller` to v20.0.0.

Adds `getNetworkClientById` callback to `TokenRatesController` constructor to [address only breaking change](https://github.com/MetaMask/core/blob/9238108438e61dcae39ec8a02d1a15aa540da4c4/packages/assets-controllers/CHANGELOG.md?plain=1#L14)

This overwrites and deletes a patch that was added on top of `@metamask/assets-controllers` version 19.0.0. This was added for the 11.7.0 extension release but can be removed now and will be added back later via https://github.com/MetaMask/core/pull/3619 (cc @DDDDDanica @Gudahtt)

### CODE REVIEW

**Review by jiexi — DISMISSED** [review]

File: yarn.lock [hunk 1]
```diff
@@ -4771,6 +4770,16 @@ __metadata:
   languageName: node
   linkType: hard
 
+"@metamask/preferences-controller@npm:^5.0.0":
```
[the remark this answers is not in the document]
Comment (adonesky1) [reply yarn.lock:4773]: Not sure I follow. This bump occurs in `@metamask/assets-controllers` v20.0.0: https://github.com/MetaMask/core/blob/ae1c9ebea94904ef0a5dd5c7c701a69a9f2c3069/packages/assets-controllers/CHANGELOG.md?plain=1#L35

**Comment by DDDDDanica** [comment]
Hey @adonesky1 thanks for the update ! i noticed the failure in pipeline probably caused by api key updates
```
      contractExchangeRates: &apos;object&apos;,
+     contractExchangeRatesByChainId: &apos;object&apos;,
```

If you do `yarn build:test` and `yarn test:e2e:single test/e2e/tests/errors.spec.js --browser=chrome --update-snapshot` should fix the snapshot

**Review by DDDDDanica — DISMISSED** [review]

File: yarn.lock (hunk 1 above)
[the remark this answers is not in the document]
Comment (adonesky1) [reply yarn.lock:4773]: 🤔 But they're different major versions? Won't `@metamask/keyring-controller@8.1.0` just continue to get `@metamask/preferences-controller@4.4.3` and now `@metamask/preferences-controller@4.4.3` will use `@metamask/preferences-controller@5.0.0`? 
We have 3 different major versions of `@metamask/eth-sig-util` in the extension's dependency graph for example. Each dep gets the right version as a sub dep:
Screenshot 2023-12-05 at 8 43 15 PM

**Review by pedronfigueiredo — APPROVED** [review]

**Review by brad-decker — APPROVED** [review]

File: yarn.lock (hunk 1 above)
[the remark this answers is not in the document]
Comment (adonesky1) [reply yarn.lock:4773]: Gotcha. Yeah I see now. Well hilariously in this case the extension uses its own version of the `PreferenceController`. And definitely no instance of the controller is passed around. The breaking change that caused the bump to version 5.0.0 was adding the `showIncomingTransactions` value to state which is never referenced anywhere in `@metamask/assets-controllers` so I think we're all good.

**Pull request merged**